mix.exs

```diff
@@ -126,39 +126,67 @@ defmodule Livebook.MixProject do
   ## Releases
 
   defp releases do
+    macos_notarization = macos_notarization()
+
     [
       livebook: [
         include_executables_for: [:unix],
         include_erts: false,
         rel_templates_path: "rel/server",
         steps: [:assemble, &remove_cookie/1]
       ],
-      mac_app: [
-        include_executables_for: [:unix],
-        include_erts: false,
-        rel_templates_path: "rel/app",
-        steps: [:assemble, &remove_cookie/1, &standalone_erlang_elixir/1, &build_mac_app/1]
-      ],
-      mac_app_dmg: [
-        include_executables_for: [:unix],
-        include_erts: false,
-        rel_templates_path: "rel/app",
-        steps: [:assemble, &remove_cookie/1, &standalone_erlang_elixir/1, &build_mac_app_dmg/1]
-      ],
-      windows_installer: [
-        include_executables_for: [:windows],
+      app: [
         include_erts: false,
         rel_templates_path: "rel/app",
         steps: [
           :assemble,
           &remove_cookie/1,
           &standalone_erlang_elixir/1,
-          &build_windows_installer/1
+          &AppBuilder.bundle/1
+        ],
+        app: [
+          name: "Livebook",
+          icon_path: [
+            macos: "rel/app/icon-macos.png",
+            windows: "rel/app/icon.ico"
+          ],
+          url_schemes: ["livebook"],
+          document_types: [
+            %{
+              name: "LiveMarkdown",
+              extensions: ["livemd"],
+              icon_path: [
+                macos: "rel/app/icon.png",
+                windows: "rel/app/icon.ico"
+              ],
+              macos_role: "Editor"
+            }
+          ],
+          additional_paths: [
+            "rel/erts-#{:erlang.system_info(:version)}/bin",
+            "rel/vendor/elixir/bin"
+          ],
+          server: LivebookApp,
+          macos_is_agent_app: true,
+          macos_build_dmg: macos_notarization != nil,
+          macos_notarization: macos_notarization,
+          windows_build_installer: true
```

Comment:
I understand you want to prefix if with the OS but it reads weird. Perhaps we should namespace it?

```
macos: [
  app_type: :regular | :agent,
  notarize: macos_notarization,
  build_dmg: macos_notarization != nil
],
windows: [
  build_installer: true
]
```

Also, IIRC, the `dmg` requires notarization or am I confused?

Reply:
I like this. Though if we follow this in :document_types it looks a bit awkward:

```elixir
document_types: [
  %{
    name: "LiveMarkdown",
    extensions: ["livemd"],
    icon_path: [
      macos: "rel/app/icon.png",
      windows: "rel/app/icon.ico"
    ],
    macos: [
      role: "Editor"
    ]
  }
}
```

doesn't it?

One way would be to allow setting _anything_ per OS:

```elixir
document_types: [
  %{
    name: "LiveMarkdown",
    extensions: ["livemd"],
    macos: [
      icon_path: "rel/app/icon.png",
      role: "Editor"
    ],
    windows: [
      icon_path: "rel/app/icon.ico"
    ]
  }
}
```

which feels like something that could be useful in the future but I also don't want to unnecessarily complicate things too early. WDYT?

Reply:
We can notarize a .dmg or a .zip (of the .app). We could just code sign (without notarising) too. To keep things simpler I chose to only codesign+notarize the dmg.